Render components to HTML

jan-cerny opened the pull request:
#### Description:

This commit adds a script `render_components.py` that renders
mappings from components to rules in HTML format. It's also
added to the pages that we publish online.

#### Rationale:


The goal is to enable easy sharing of component mappings
outside of the team and simplify the review of component rules
for content developers.


#### Review Hints:
build the content, and then run
python3 utils/render_components.py $(pwd) /tmp/components
firefox /tmp/components/index.html

jan-cerny commented:
I have rebased this PR on the top of the latest upstream master branch. 

I have some additional thoughts:

* To show meaningful content in the rules section we need to use resolved rules but these are different for each product therefore I have decided to generate the content mapping per-product. Do you think that the current form is useful? Would it be better to have only one page for each component and show all the products and rules in product in that single page instead?
* It could be useful to have a module for a nice rendering of rule and rule details. We already have multiple parts of code that deals with that but isn't reusable. What do you think?

Mab879 reviewed (changes requested):
Thanks for the PR please take look at my comments.

jan-cerny commented on tests/unit/ssg-module/data/components_dir/fapolicyd.yml:
```diff
@@ -10,3 +10,5 @@ rules:
 - service_fapolicyd_enabled
 templates:
 - file_policy_blocked
+changelog:
```
Yes, it's at https://complianceascode.readthedocs.io/en/latest/manual/developer/03_creating_content.html#components

jan-cerny commented:
I have created a wrapper function to avoid using the private method, I have used long variable names, changed to use the `--root` option, add typing hints, changed case title, renamed variables and quote bash variables.
